File: munoz-mqtt-switch-neopixel.py
from adafruit_circuitplayground import cp
import paho.mqtt.client as mqtt

def on_connect(client, userdata, flags, rc): # is used for checking that the connection was established 
    if rc == 0: #if the rc is equal to 0
        cp.red_led = True # the led in index 0 will on/light 
        client.subscribe("switch/#") # find the the published topic

def on_message(client, userdata, msg): #is used for checking published topic is subscribe  
    if msg.topic=="switch/switch0": #if the topic is equal to the subscribe topic in which the switch 0 is on 
        if msg.payload.decode() == "true": #and the msg.payload.decode is equal
            cp.pixels[0] = (255, 0, 0) #the pixel 0 will turn on in red color
        else: #if not
            cp.pixels[0] = (0, 0, 0) #the pixel 0 is off
    elif msg.topic=="switch/switch1": #if the topic is equal to the subscribe topic in which the switch 1 is on 
        if msg.payload.decode() == "true":#and the msg.payload.decode is equal
            cp.pixels[1] = (0, 255, 0)#the pixel 1 will turn on in green color
        else:
            cp.pixels[1] = (0, 0, 0)#the pixel 0 is off
    elif msg.topic=="switch/switch2": #if the topic is equal to the subscribe topic in which the switch 2 is on 
        if msg.payload.decode() == "true":#and the msg.payload.decode is equal
            cp.pixels[2] = (0, 0, 255) #the pixel 2 will turn on in blue color
        else:
            cp.pixels[2] = (0, 0, 0)#the pixel 0 is off
cp.red_led = False

client = mqtt.Client()
client.on_connect = on_connect
client.on_message = on_message

client.connect("test.mosquitto.org", 1883, 60) #connection address

client.loop_forever()

# Broker for online client: https://iamelijah2016.github.io/
# wss://mqtt.eclipse.org:443/mqtt
# wss://test.mosquitto.org:8081/mqtt

File: munoz-mqtt-switch-neopixel.js
// var client  = mqtt.connect({ host:'test.mosquitto.org', port: 8081})
// or
// var client  = mqtt.connect('wss://test.mosquitto.org:8081/mqtt')

// var client  = mqtt.connect({ host:'mqtt.eclipse.org/mqtt', port: 443})
// or
// var client  = mqtt.connect('wss://mqtt.eclipse.org:443/mqtt')

var broker = 'wss://test.mosquitto.org:8081/mqtt'; //connection address
var client  = mqtt.connect(broker);

var status_header = document.getElementById('status-header')

client.on('connect', function () {
    status_header.innerHTML = 'Connected to '+broker; 
    console.log('Connected to '+broker)
})
//switch 0
var pub_switch_0 = document.getElementById('switch0');
pub_switch_0.onclick = () => {
    console.log(pub_switch_0.checked)
    client.publish('switch/switch0', String(pub_switch_0.checked))
}
//switch1
var pub_switch_1 = document.getElementById('switch1');
pub_switch_1.onclick = () => {
    console.log(pub_switch_1.checked)
    client.publish('switch/switch1', String(pub_switch_1.checked))
}
//switch
var pub_switch_2 = document.getElementById('switch2');
pub_switch_2.onclick = () => {
    console.log(pub_switch_2.checked)
    client.publish('switch/switch2', String(pub_switch_2.checked))
}


